### DISCUSSION TITLE
Typing Errors

### DISCUSSION THREAD

**Discussion by ghost** [Asjas/prisma-redis-middleware#148] [Q&A] [answered]
Hello,

1. here is an error with the `Middleware` type returned by `createPrismaRedisCache`

```
Types of parameters 'params' and 'params' are incompatible.

Property 'model' is optional in type 'MiddlewareParams' but required in type 'MiddlewareParams'
```

![Screenshot 2022-05-26 at 19 52 37](https://user-images.githubusercontent.com/8101584/170546716-63949b39-d457-4952-bb43-2fe4ae100622.jpg)

can be resolved by:

![Screenshot 2022-05-26 at 19 54 08](https://user-images.githubusercontent.com/8101584/170546965-130fb4ff-f060-410d-8396-f2bee0cc8eda.jpg)

2. and the redis client:

```
Type 'RedisOptions' is not assignable to type 'CommonRedisOptions'.
The types returned by 'new Connector(...)' are incompatible between these types.
Property 'connecting' is protected but type 'AbstractConnector' is not a class derived from 'AbstractConnector'.
```

![Screenshot 2022-05-26 at 19 55 33](https://user-images.githubusercontent.com/8101584/170547406-0e77565f-5a67-40c8-99f5-ab5c1b66b1cb.jpg)

I am using with `nestjs` and inject the redis client from `ioredis` by `@liaoliaots/nestjs-redis`, but it has the same error with direct `const redis = new Redis(); `

**Comment by Asjas** [comment, marked as answer]
@lodisy I changed these types in release 4.0.0 - can you test if they work now?

**Reply by ghost** [reply]
sorry I have stopped using Prisma anymore
